### PR TITLE
Cleanup PR build pipelines

### PR DESCRIPTION
**Pull request by hsubramanianaks**
This PR is to clean up PR build pipelines and make them simple. 

image

### CODE REVIEW

File: build-pipelines-PR.yml [hunk 1]
```diff
@@ -103,7 +69,6 @@ jobs:
       restoreArguments: -r win-x64
       selectOrConfig: config
       nugetConfigPath: src/nuget.config
-      externalEndpoints: dba0ba50-2d4e-4f12-9f5a-42a638da803b
```
Comment (elenavillamil) [inline build-pipelines-PR.yml:106]: is this removal intentional, if so why is this not needed now?

File: build-pipelines-PR.yml [hunk 2]
```diff
@@ -129,7 +93,7 @@ jobs:
     enabled: True
     inputs:
       solution: src/client.sln
-      msbuildArgs: -p:Localize=true -p:ConsumeEndpointManager=false
```
Comment (elenavillamil) [inline build-pipelines-PR.yml:132]: why remove this part? This taks is the one failing when PR is sent from fork, is this the reason why?

File: build-pipelines-PR.yml [hunk 3]
```diff
@@ -225,224 +151,6 @@ jobs:
...
-
-
-        New-Item -Path './src/dsc/bin/${{ parameters.BuildConfiguration }}/net6.0/linux-x64/publish/kubectl' -ItemType Directory
-
-        New-Item -Path './src/dsc/bin/${{ parameters.BuildConfiguration }}/net6.0/linux-x64/publish/kubectl/linux' -ItemType Directory
-
-        curl https://dl.k8s.io/release/v1.21.2/bin/linux/amd64/kubectl  -OutFile ./src/dsc/bin/${{ parameters.BuildConfiguration }}/net6.0/linux-x64/publish/kubectl/linux/kubectl
-
-  - task: CopyFiles@2
-    displayName: Collect files for .zip (Windows)
```
Comment (elenavillamil) [inline build-pipelines-PR.yml:256]: Simpler is great, but for some of these tasks I wonder if it is worth keeping parity with release pipeline. So PR build and release pipeline stay as close as possible aside from tasks that cant be ran from fork. I am open either way, just thinking.

Comment (hsubramanianaks) [reply build-pipelines-PR.yml:256]: This will reduce the time PR builds take and it is already validating if it restores, builds and UT's. I think these copy files, zip tasks wouldn't change much. if you think if we want to keep parity then we can't save much on the build time. Just a suggestion on pros/cons of doing it simple.

File: build-pipelines-PR.yml (hunk 2 above)
Comment (hsubramanianaks) [reply build-pipelines-PR.yml:132]: so on the other parameter ConsumeEndpointManger is false because we are building/publishing endpoint manager as separate step in the pipelines so that main dsc project doesn't do it again. By default value is true. 

image

Comment (hsubramanianaks) [reply build-pipelines-PR.yml:132]: The failures for the forked repos are related to the external endpoints being present in the restore endpoint manager step above https://github.com/Azure/Bridge-To-Kubernetes/pull/193/files#r1131937016. That's why this step is failing, please see below screenshot. 

image

image

File: build-pipelines-PR.yml (hunk 1 above)
Comment (hsubramanianaks) [reply build-pipelines-PR.yml:106]: here is the reference to the document 
image

File: build-pipelines-PR.yml (hunk 3 above)
Comment (elenavillamil) [reply build-pipelines-PR.yml:256]: sounds good. let go with removing them.

File: build-pipelines-PR.yml [hunk 4]
```diff
@@ -452,125 +160,54 @@ jobs:
...
+  displayName: Run PR build for non self contained
   timeoutInMinutes: 120
   cancelTimeoutInMinutes: 1
   pool:
     name: 'VSEngSS-MicroBuild2022-1ES'
     demands:
     - msbuild
     - visualstudio
-    - vstest
+    - DotNetFramework
```
Comment (elenavillamil) [inline build-pipelines-PR.yml:173]: intentional? Curious why.

File: build-pipelines-PR.yml [hunk 5]
```diff
@@ -51,7 +40,7 @@ resources:
     ref: refs/heads/main
 jobs:
 - job: Phase_1
-  displayName: Run self-contained build (binaries V1, where the binaries are all bundled together)
+  displayName: Run PR build for self contained
```
Comment (elenavillamil) [inline build-pipelines-PR.yml:43]: nit and maybe just my personal reference but i like having the link between self contained and binaries V1. If possible I liked previous display name better. okay to update display name if we add a comment as well.

**Review by elenavillamil — APPROVED** [review]
Looks good. Left two minor comments.

File: build-pipelines-PR.yml (hunk 4 above)
Comment (hsubramanianaks) [reply build-pipelines-PR.yml:173]: I kept it similar to self-contained one's if you see above, it has all these three demands.

File: build-pipelines-PR.yml (hunk 5 above)
Comment (hsubramanianaks) [reply build-pipelines-PR.yml:43]: I changed it back, since it was long, and it was cutting off in the ADO UI that's why made it short.

File: build-pipelines-PR.yml (hunk 4 above)
Comment (elenavillamil) [reply build-pipelines-PR.yml:173]: ok we may want to update release pipeline too then?

**Review by elenavillamil — APPROVED** [review]
LGMT
